src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -320,6 +319,26 @@ public List<Course> getAllCourses(@RequestParam(defaultValue = "false") boolean
         return userCourses.toList();
     }
 
+    /**
+     * GET /courses/groups : get all groups for all courses for administration purposes.
+     *
+     * @return the list of groups (the user has access to)
+     */
+    @GetMapping("courses/groups")
+    @PreAuthorize("hasRole('TA')")
+    public ResponseEntity<List<String>> getAllGroupsForAllCourses() {
```

Comment:
You should return a set here, since different courses can have the same group names

Reply:
resolved